src/grammar.coffee

```diff
@@ -166,8 +166,8 @@ grammar =
   ]
 
   String: [
-    o 'STRING',                                 -> new StringLiteral $1
-    o 'STRING_START Body STRING_END',           -> new StringWithInterpolations $2
+    o 'STRING',                                 -> new StringLiteral $1, {quote: $1.quote, initialChunk: $1.initialChunk, finalChunk: $1.finalChunk, indent: $1.indent, double: $1.double}
```

Comment:
> So this is the method for passing extra data through the parser? Clever, I can get behind this.
> This particular object is ambiguous though, it should have braces.

@GeoffreyBooth yup credit to whoever figured out how to expose `.generated` to the grammar, that's where I got the pattern of wrapping in a `new String()`

Ok added explicit braces - for the sake of my thinking about Prettier formatting, can you try and explain why you consider this trailing object ambiguous while the one on the next line (`quote: $1.quote`) is ok?

Reply:
The next line isn’t great either 😉 Technically the next line isn’t ambiguous because it doesn’t have any commas: it’s an object with just one key. The line I objected to was an inline object with multiple key-value pairs separated by commas, and it’s ambiguous (to a human reader, at least) whether those commas refer to the object or to arguments of the `new StringLiteral` constructor/function call. But I’m not a big fan of implicit inline objects in function calls, there’s just so much already going on with a function call that braces are generally helpful here.